_posts/2020-12-18-3-dimensional-segmentation.md

```diff
@@ -0,0 +1,88 @@
+
+```python
+def save_found_slices(fstc_slices):
+    fstc_slices = fstc_slices[0]
+    fstc_ndarray = slice_image_list_to_ndarray(fstc_slices)
+    # if the tensor is all zero (no partitions found) return None
+    if np.all(fstc_ndarray == 0):
+        return None
+    # from https://gist.github.com/arokem/423d915e157b659d37f4aded2747d2b3
+    fstc_nii = nib.Nifti1Image(skimage.img_as_ubyte(fstc_ndarray), affine=None)
+    fstcbytes = io.BytesIO()
+    file_map = fstc_nii.make_file_map({"image": fstcbytes, "header": fstcbytes})
+    fstc_nii.to_file_map(file_map)
+    fstcb64 = base64.b64encode(fstcbytes.getvalue()).decode()
+    return fstcb64
+```
+
+A clientside callback fires when the base64 string is written to the `Store`, decodes the string into a [`Blob`](https://developer.mozilla.org/en-US/docs/Web/API/Blob) and writes to the `href` property of a hidden `Div` as a "download link". The final step is another clientside callback that fires when the `href` of the hidden `Div` is updated and simulates a "click" by the user on this link. 
+
+```javascript
+app.clientside_callback(
+    """
+function (href) {
+    if (href != "") {
+        let download_a=document.getElementById("download-link");
+        download_a.click();
+    }
+    return '';
+}
+""",
+    Output("dummy", "children"),
+    [Input("download-link", "href")],
+)
+```
+
+This opens the download interface of the users browser where the blob can be saved to disk.
+
+## What's next
+
+We hope you like this app for volumetric image annotation with superpixels. If you have suggestions, you can submit them via the [dash-sample-apps repository](https://github.com/plotly/dash-sample-apps) or on [Twitter](https://twitter.com/EGouillart). If you have support questions, the best place for them is the Plotly [community forum](https://community.plotly.com/). 
+
+If you want to annotate and explore 3D volumetric image data, you can use this app as a starting 
+point and adapt it for your own needs. You might also want to check out the other [image processing apps in the dash app gallery](https://dash-gallery.plotly.host/Portal/?search=[Image%20Processing]). Some additional resources:
+
+- the new [`Dash-Slicer` component](https://github.com/plotly/dash-slicer) that makes it [very easy to build interactive 3D image slicers](https://dash.plotly.com/slicer) in Dash in only a few lines of code
+- the new [functionality for image annotation in Dash](https://dash.plotly.com/annotations) and the corresponding [Blogpost](https://eoss-image-processing.github.io/2020/05/06/shape-drawing.html)
```

Comment:
Could you please also link to the chest X-ray app, and the corresponding blog post, for another 3-D app?